java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -185,23 +188,33 @@ public boolean equals(Object obj) {
     }
     MethodYield other = (MethodYield) obj;
     return new EqualsBuilder()
-      .append(parametersConstraints, other.parametersConstraints)
+      .append(parametersConstraints.stream().flatMap(MethodYield::pmapToStream).collect(Collectors.toList()),
```

Comment:
because of the pmapToStream method returning an empty stream in case of 'no constraint' for a parameter, it means that we can't differentiate call to `foo(Object a, Object b, Object c, Object d)` with the following constraints:
*  foo([NOT_NULL],[NOT_NULL],[],[])
*  foo([NOT_NULL],[],[],[NOT_NULL])